File: Solutions/[02]_1.py
n = int(input())

count = 1
stack = []
result = []

for i in range(1, n + 1): # 데이터 개수만큼 반복
    data = int(input())
    while count <= data: # 입력 받은 데이터에 도달할 때까지 삽입
        stack.append(count)
        count += 1
        result.append('+')
    if stack[-1] == data: # 스택의 최상위 원소가 데이터와 같을 때 출력
        stack.pop()
        result.append('-')
    else: # 불가능한 경우
        print('NO')
        exit(0)

print('\n'.join(result)) # 가능한 경우

File: Solutions/[02]_1.cpp
#include <bits/stdc++.h>

using namespace std;

int n, cnt = 1;
stack<int> s;
vector<char> result;

int main() {
	cin >> n;
	for (int i = 0; i < n; i++) {
		int x;
		cin >> x;
		// 입력 받은 데이터에 도달할 때까지 삽입 
		while (cnt <= x) {
			s.push(cnt);
			cnt += 1;
			result.push_back('+');
		}
		// 스택의 최상의 원소가 데이터와 같을 때 출력 
		if (s.top() == x) {
			s.pop();
			result.push_back('-');
		}
		else {
			cout << "NO";
			return 0;
		}
	}
	// 결과를 차례대로 출력 
	for (auto x: result) {
		cout << x << '\n';
	}
}
